### PR TITLE
MacOS Qt5 overlay clean up

### PR DESCRIPTION
**Pull request by daschuer**
This is a set of patches that improves our Qt5.12 backport for MacOs

### CODE REVIEW

File: overlay/osx/qt5-base/portfile.cmake [hunk 1]
```diff
@@ -173,6 +173,12 @@ elseif(VCPKG_TARGET_IS_OSX)
             "QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.12" 
             "QMAKE_MACOSX_DEPLOYMENT_TARGET = ${VCPKG_OSX_DEPLOYMENT_TARGET}" 
             _tmp_contents "${_tmp_contents}")
+        if("${VCPKG_TARGET_ARCHITECTURE}" MATCHES "arm64")
+            string(REPLACE 
+                "QMAKE_APPLE_DEVICE_ARCHS = x86_64" 
+                "QMAKE_APPLE_DEVICE_ARCHS = arm64" 
+                _tmp_contents ${_tmp_contents})
+        endif()
```
Comment (Swiftb0y) [inline overlay/osx/qt5-base/portfile.cmake:181]: can you explain this? Is this a change from us or a patch from Qt5.15?

Comment (daschuer) [reply overlay/osx/qt5-base/portfile.cmake:181]: Ups, this slips in from the M1 arm port.
I must have messed up the rebase. 
I will rebase this again to remove that change.

**Comment by daschuer** [comment]
Done.

**Review by Swiftb0y — APPROVED** [review]
LGTM

**Pull request merged**